R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+  while(length(line) != 0){
+    full_info <- strsplit(line, ":")[[1]]
+    
+    if(length(full_info) != length(col_names)+1)
+      stop("Missing attributes/values in series.")
+    
+    series <- strsplit(tail(full_info, 1), ",")[[1]]
+    series[which(series == "?")] <- NA
+    series <- as.numeric(series)
+    
+    if(sum(is.na(series)) == length(series))
+      stop("All series values are missing. A given series should contains a set of comma separated numeric values. At least one numeric value should be there in a series.")
+    
+    values <- c(values, series)
+    row_count <- row_count + length(series)
+    
+    attributes <- head(full_info, length(full_info)-1)
+    
+    for(col in seq_along(col_names)){
+      
+      att <- eval(parse(text=col_names[col]))
+      
+      # This format supports 3 attribute types: string, numeric and date
+      if(col_types[col] == "date"){
+        if(is.null(frequency))
+          stop("Frequency is missing.")
+        else{
+          if(frequency %in% LOW_FREQUENCIES)
+            start_time <- as.POSIXct(attributes[col], format = "%Y-%m-%d %H-%M-%S")
+          else if(frequency %in% HIGH_FREQUENCIES)
+            start_time <- as.Date(attributes[col], format = "%Y-%m-%d %H-%M-%S")
+          else
+            stop("Invalid frequency.")
+          
+          if(is.na(start_time))
+            stop("Incorrect timestamp format. Specify your timestamps as YYYY-mm-dd HH-MM-SS")
+        }
+        
+        att <- append(att, seq(start_time, length = length(series), by = FREQ_MAP[[frequency]]))
+      }else{
+        if(col_types[col] == "numeric")
+          attributes[col] <- as.numeric(attributes[col])
+        else if(col_types[col] == "string")
+          attributes[col] <- as.character(attributes[col])
+        else
+          stop("Invalid attribute type.")
+        
+        if(is.na(attributes[col]))
+          stop("Invalid attribute values.")
+        
+        att <- append(att, rep(attributes[col], length(series)))
+      }
+      assign(col_names[col], att)
+    }
+    
+    line <- readLines(file, n = 1)
+  }
+  
+  data <- as.data.frame(matrix(nrow = row_count, ncol = length(col_names) + 1))
+  colnames(data) <- c(col_names, value_column_name)
+  
+  for(col in col_names)
+    data[[col]] <- eval(parse(text = col))
+  
+  data[[value_column_name]] <- values
+  
+  if(!(is.null(key))){
+    if(!(key %in% col_names))
+      stop("Invalid key. Cannot convert data into tsibble format.")
+    else{
+      if(is.null(index)){
+        if(is.null(index_var))
+          cat("Index is not provided. No valid index found in data. Returning a data.frame.")
+        else
+          data <- tsibble::build_tsibble(x = data, key = key, index = index_var, ordered = F)
+      }else{
+        if(!(index %in% col_names))
+          stop("Invalid index Cannot convert data into tsibble format.")
+        else
+          data <- tsibble::build_tsibble(x = data, key = key, index = index, ordered = F)
```

Comment:
Use `FALSE` instead of `F` for best practices.

Reply:
Fixed as suggested